tests/modcc/test_kinetic_rewriter.cpp

```diff
@@ -0,0 +1,507 @@
+    using namespace alg;
+
+    if (const auto& n = e->is_number()) {
+        return prodsum(n->value());
+    }
+    else if (const auto& c = e->is_function_call()) {
+        std::stringstream rep(c->name());
+        rep << '(';
+        bool first = true;
+        for (const auto& arg: c->args()) {
+            if (!first) rep << ',';
+            rep << expand_expression(arg.get(), exmap);
+            first = false;
+        }
+        rep << ')';
+        return prodterm(rep.str());
+    }
+    else if (const auto& i = e->is_identifier()) {
+        std::string k = i->spelling();
+        auto x = exmap.find(k);
+        //return x!=exmap.end()? x->second: prodterm(k);
+        auto y =  x!=exmap.end()? x->second: prodterm(k);
+        if (k=="v") {
+            std::cout << "debug: y=" << y << "\n";
+        }
+        return y;
+    }
+    else if (const auto& b = e->is_binary()) {
+        prodsum lhs = expand_expression(b->lhs(), exmap);
+        prodsum rhs = expand_expression(b->rhs(), exmap);
+
+        switch (b->op()) {
+        case tok::plus:
+            return lhs+rhs;
+        case tok::minus:
+            return lhs-rhs;
+        case tok::times:
+            return lhs*rhs;
+        case tok::divide:
+            return lhs/rhs;
+        case tok::pow:
+            if (!rhs.is_scalar()) {
+                // make an opaque term for this case (i.e. too hard to simplify)
+                return prodterm("("+to_string(lhs)+")^("+to_string(rhs)+")");
+            }
+            else return pow_expand(lhs, rhs.first_coeff());
+        default:
+            throw std::runtime_error("unrecognized binop");
+        }
+    }
+    else if (const auto& u = e->is_unary()) {
+        prodsum inner = expand_expression(u->expression(), exmap);
+        switch (u->op()) {
+        case tok::minus:
+            return -inner;
+        case tok::exp:
+            return prodterm("exp("+to_string(inner)+")");
+        case tok::log:
+            return prodterm("log("+to_string(inner)+")");
+        case tok::sin:
+            return prodterm("sin("+to_string(inner)+")");
+        case tok::cos:
+            return prodterm("cos("+to_string(inner)+")");
+        default:
+            throw std::runtime_error("unrecognized binop");
+        }
+    }
+    else {
+        throw std::runtime_error("unexpected expression type");
+    }
+}
+
+std::map<std::string, alg::prodsum> expand_assignments(stmt_list_type& stmts) {
+    using namespace alg;
+
+    // This is 'just a test', so don't try to be complete: functions are
+    // left unexpanded; procedure calls are ignored.
+
+    std::map<std::string, prodsum> exmap;
+
+    for (const auto& stmt: stmts) {
+        if (auto assign = stmt->is_assignment()) {
+            auto lhs = assign->lhs();
+            std::string key;
+            if (auto deriv = lhs->is_derivative()) {
+                key = deriv->spelling()+"'";
+            }
+            else if (auto id = lhs->is_identifier()) {
```

Comment:
It should be taken in the e.g. the analysis of the example `DERIVATIVE` block, where in one instance we have an assignment to `a'` (the first path) and two lines below, an assignment to `rev2` (the second path).